src/instructorTabs/InstructorTabs.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect, useCallback, useContext } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { Tab, Tabs } from '@openedx/paragon';
+import { SlotContext, BaseSlotOperation, useSlotOperations } from '@openedx/frontend-base';
+
+export interface TabProps {
+  tab_id: string,
+  url: string,
+  title: string,
+}
+
+interface SlotWithElementOperation extends BaseSlotOperation {
+  element: React.ReactElement,
+}
+
+const InstructorTabs = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const { id: slotId } = useContext(SlotContext);
+  const widgets: SlotWithElementOperation[] = useSlotOperations(slotId) as SlotWithElementOperation[];
+
+  const [tabKey, setTabKey] = useState<string>('courseInfo');
+
+  const getActiveTabFromUrl = useCallback(() => {
+    const currentPath = location.pathname.split('/').pop() ?? '';
+    const activeTab = widgets.find((slot) => slot.element.props.url === currentPath)?.element;
```

Comment:
This assumes that the active tab is always the last component of the /slash/separated/url/path/, but in the future there may be additional URLs within each tab, e.g. to open a modal dialog or have sub-tabs. So I recommend using `useParams` instead to get the named parameter from the URL instead of manually splitting the URL into parts and taking the last part.